packages/stream-api/src/time.ts

```diff
@@ -0,0 +1,71 @@
+import { google } from '@vertexvis/frame-streaming-protos';
+import Long from 'long';
+
+type MaybeTimestampOrDuration = Pick<
+  google.protobuf.ITimestamp,
+  'seconds' | 'nanos'
+>;
+
+type DefinedTimestampOrDuration = Required<
+  Pick<google.protobuf.ITimestamp, 'seconds' | 'nanos'>
+>;
+
+/**
+ * Converts a JS date type to a `google.protobuf.Timestamp`. The returned time
+ * will be represented as UTC.
+ *
+ * @param date The date to convert.
+ */
+export function toProtoTimestamp(date: Date): google.protobuf.Timestamp {
+  const millis = date.getTime();
+  return google.protobuf.Timestamp.create(parseEpochMillis(millis));
+}
+
+/**
+ * Returns the current date and time as a `google.protobuf.Timestamp`.
+ */
+export function currentDateAsProtoTimestamp(): google.protobuf.Timestamp {
+  return toProtoTimestamp(new Date());
+}
+
+/**
+ * Converts the two dates into a `google.protobuf.Duration`, where the duration
+ * is `end - start`. If `end` is before `start`, then the duration will be
+ * negative.
+ *
+ * @param start The starting time.
+ * @param end The ending time.
+ */
+export function toProtoDuration(
+  start: Date,
+  end: Date
+): google.protobuf.Duration {
+  const millis = end.getTime() - start.getTime();
+  return google.protobuf.Duration.create(parseEpochMillis(millis));
+}
+
+/* eslint-disable padding-line-between-statements */
```

Comment:
Do we need to disable the eslint rules here? Would prefer to block the rule as a whole if we're going to add these disables in files.

Reply:
Without this the method overloads are not grouped. IMO it's clearer that these are overloaded methods without whitespace between them.

Method overloads are fairly rare, so I think it's okay to add in these exceptions vs globally disabling the lint rule. I'll usually consider removing the lint rule only if I find myself adding an exception every few days.

Reply:
Sounds good to me. Thanks for the explanation.